### ISSUE TITLE
bug(matierl/datepicker): Voiceover does not announce current month after clicking next month button

### ISSUE THREAD

**Issue by zarend** [angular/components#24397]
### Is this a regression?

- AFAIK no

### Description

When using the datepicker with voiceover the month is not announced after clicking the month/previous month buttons.

### Reproduction

Steps to reproduce:
1.  With screenreader enamed go to [basic Datepicker example](https://material.angular.io/components/datepicker/overview#datepicker-overview)
2. Open the calendar and navigate to Feb 1, 2022
3. Click the next month button
4. (Calendar displays the month of March 2022 and voiceover does not announce anything)

### Expected Behavior

Screenreader should announce Mar 2022

### Actual Behavior

screenreader does not announce anything after click the "Next Month" button

### Environment

MacOS 12.2 (21D49)
Chrome Version 98.0.4758.80 (Official Build) (x86_64)
Safari Version 15.3 (17612.4.9.1.5)
Voiceover
